// @callstack/reassure-cli/lib/commonjs/commands/measure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.command = void 0;
exports.run = run;
var _fs = require("fs");
var _path = require("path");
var _child_process = require("child_process");
var _reassureCompare = require("@callstack/reassure-compare");
var _git = require("../utils/git");
const RESULTS_DIRECTORY = '.reassure';
const RESULTS_FILE = '.reassure/current.perf';
const BASELINE_FILE = '.reassure/baseline.perf';
async function run(options) {
  const measurementType = options.baseline ? 'Baseline' : 'Current';
  const metadata = {
    branch: (options === null || options === void 0 ? void 0 : options.branch) ?? (await (0, _git.getGitBranch)()),
    commitHash: (options === null || options === void 0 ? void 0 : options.commitHash) ?? (await (0, _git.getGitCommitHash)())
  };
  console.log(`\n❇️  Running performance tests:`);
  console.log(` - ${measurementType}: ${(0, _reassureCompare.formatMetadata)(metadata)}\n`);
  (0, _fs.mkdirSync)(RESULTS_DIRECTORY, {
    recursive: true
  });
  const outputFile = options.baseline ? BASELINE_FILE : RESULTS_FILE;
  (0, _fs.rmSync)(outputFile, {
    force: true
  });
  const header = {
    metadata
  };
  (0, _fs.writeFileSync)(outputFile, JSON.stringify(header) + '\n');
  const defaultPath = process.platform === 'win32' ? 'node_modules/jest/bin/jest' : 'node_modules/.bin/jest';
  const testRunnerPath = process.env.TEST_RUNNER_PATH ?? defaultPath;
  const defaultArgs = '--runInBand --testMatch "<rootDir>/**/*.perf-test.[jt]s?(x)"';
  const testRunnerArgs = process.env.TEST_RUNNER_ARGS ?? defaultArgs;
  const spawnInfo = (0, _child_process.spawnSync)('node', ['--jitless', '--expose-gc', '--no-concurrent-sweeping', '--max-old-space-size=4096', testRunnerPath, testRunnerArgs], {
    shell: true,
    stdio: 'inherit',
    env: {
      ...process.env,
      OUTPUT_FILE: outputFile
    }
  });
  console.log('');
  if (spawnInfo.status !== 0) {
    console.error(`❌  Test runner (${testRunnerPath}) exited with error code ${spawnInfo.status}`);
    process.exitCode = 1;
    return;
  }
  if ((0, _fs.existsSync)(outputFile)) {
    console.log(`✅  Written ${measurementType} performance measurements to ${outputFile}`);
    console.log(`🔗 ${(0, _path.resolve)(outputFile)}\n`);
  } else {
    console.error(`❌  Something went wrong, ${measurementType} performance file (${outputFile}) does not exist\n`);
    return;
  }
  if (options.baseline) {
    console.log("Hint: You can now run 'reassure' to measure & compare performance against modified code.\n");
    return;
  }
  if (options.compare) {
    if ((0, _fs.existsSync)(BASELINE_FILE)) {
      (0, _reassureCompare.compare)();
    } else {
      console.log(`Baseline performance file does not exist, run 'reassure --baseline' on your baseline code branch to create it.\n`);
      return;
    }
  }
}
const command = {
  command: ['measure', '$0'],
  describe: 'Gather performance measurements by running performance tests',
  builder: yargs => {
    return yargs.option('baseline', {
      type: 'boolean',
      default: false,
      describe: 'Save measurements as baseline instead of current'
    }).option('compare', {
      type: 'boolean',
      default: true,
      describe: 'Outputs performance comparison results'
    }).option('branch', {
      type: 'string',
      describe: 'Branch name of current code to be included in the report'
    }).option('commit-hash', {
      type: 'string',
      describe: 'Commit hash of current code to be included in the report'
    });
  },
  handler: args => run(args)
};
exports.command = command;
