// @callstack/reassure-compare/lib/commonjs/compare.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.compare = compare;
exports.loadFile = loadFile;
var fsSync = _interopRequireWildcard(require("fs"));
var _console = require("./output/console");
var _json = require("./output/json");
var _markdown = require("./output/markdown");
var _logs = require("./utils/logs");
var _validate = require("./utils/validate");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/**
 * Probability threshold for considering given difference signficiant.
 */
const PROBABILITY_CONSIDERED_SIGNIFICANT = 0.02;

/**
 * Render duration threshold (in ms) for treating given difference as significant.
 *
 * This is additional filter, in addition to probability threshold above.
 * Too small duration difference might be result of measurement grain of 1 ms.
 */
const DURATION_DIFF_THRESHOLD_SIGNIFICANT = 4;

/**
 * Threshold for considering render count change as significant. This implies inclusion
 * of scenario results in Render Count Changed output section.
 */
const COUNT_DIFF_THRESHOLD = 0.5;
/**
 * Main routine.
 *
 * Responsible for loading baseline and current performance results and outputting data in various formats.
 */
function compare() {
  let {
    baselineFile = '.reassure/baseline.perf',
    currentFile = '.reassure/current.perf',
    outputFile = '.reassure/output.json',
    outputFormat = 'all'
  } = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  const hasCurrentFile = fsSync.existsSync(currentFile);
  if (!hasCurrentFile) {
    (0, _logs.logError)(`Current results files "${currentFile}" does not exists. Check your setup.`);
    process.exit(1);
  }
  let currentResults;
  try {
    currentResults = loadFile(currentFile);
  } catch (error) {
    (0, _logs.logError)(`Error while parsing file: ${currentFile}`, error);
    process.exit(1);
  }
  const hasBaselineFile = fsSync.existsSync(baselineFile);
  if (!hasBaselineFile) {
    (0, _logs.logWarning)(`Baseline results files "${baselineFile}" does not exists. This warning should be ignored only if you are bootstapping perf test setup, otherwise it indicates invalid setup.`);
  }
  let baselineResults = null;
  if (hasBaselineFile) {
    try {
      baselineResults = loadFile(baselineFile);
    } catch (error) {
      (0, _logs.logError)(`Error while parsing file: ${baselineFile}`, error);
      process.exit(1);
    }
  }
  const output = compareResults(currentResults, baselineResults);
  if (outputFormat === 'console' || outputFormat === 'all') {
    (0, _console.printToConsole)(output);
  }
  if (outputFormat === 'json' || outputFormat === 'all') {
    (0, _json.writeToJson)(outputFile, output);
  }
  if (outputFormat === 'markdown' || outputFormat === 'all') {
    (0, _markdown.writeToMarkdown)('.reassure/output.md', output);
  }
}

/**
 * Load performance file and parse it to `PerformanceRecord` object.
 */
function loadFile(path) {
  var _header;
  const contents = fsSync.readFileSync(path, 'utf8');
  const lines = contents.split(/\r?\n/).filter(line => !!line.trim()).map(line => JSON.parse(line));
  let header = null;
  let entries = [];
  const hasHeader = lines[0].metadata !== undefined;
  if (hasHeader) {
    header = (0, _validate.parseHeader)(lines[0]);
    entries = (0, _validate.parsePerformanceEntries)(lines.slice(1));
  } else {
    entries = (0, _validate.parsePerformanceEntries)(lines);
  }
  const keyedEntries = {};
  entries.forEach(entry => {
    keyedEntries[entry.name] = entry;
  });
  return {
    metadata: (_header = header) === null || _header === void 0 ? void 0 : _header.metadata,
    entries: keyedEntries
  };
}

/**
 * Compare results between baseline and current entries and categorize.
 */
function compareResults(current, baseline) {
  // Unique test scenario names
  const names = [...new Set([...Object.keys(current.entries), ...Object.keys((baseline === null || baseline === void 0 ? void 0 : baseline.entries) || {})])];
  const compared = [];
  const added = [];
  const removed = [];
  names.forEach(name => {
    const currentEntry = current.entries[name];
    const baselineEntry = baseline === null || baseline === void 0 ? void 0 : baseline.entries[name];
    if (currentEntry && baselineEntry) {
      compared.push(buildCompareEntry(name, currentEntry, baselineEntry));
    } else if (currentEntry) {
      added.push({
        name,
        current: currentEntry
      });
    } else if (baselineEntry) {
      removed.push({
        name,
        baseline: baselineEntry
      });
    }
  });
  const significant = compared.filter(item => item.isDurationDiffSignificant).sort((a, b) => b.durationDiff - a.durationDiff);
  const meaningless = compared.filter(item => !item.isDurationDiffSignificant).sort((a, b) => b.durationDiff - a.durationDiff);
  const countChanged = compared.filter(item => Math.abs(item.countDiff) > COUNT_DIFF_THRESHOLD).sort((a, b) => b.countDiff - a.countDiff);
  added.sort((a, b) => b.current.meanDuration - a.current.meanDuration);
  removed.sort((a, b) => b.baseline.meanDuration - a.baseline.meanDuration);
  return {
    metadata: {
      current: current.metadata,
      baseline: baseline === null || baseline === void 0 ? void 0 : baseline.metadata
    },
    errors: _logs.errors,
    warnings: _logs.warnings,
    significant,
    meaningless,
    countChanged,
    added,
    removed
  };
}

/**
 * Establish statisticial signficance of render duration difference build compare entry.
 */
function buildCompareEntry(name, current, baseline) {
  const durationDiff = current.meanDuration - baseline.meanDuration;
  const relativeDurationDiff = durationDiff / baseline.meanDuration;
  const countDiff = current.meanCount - baseline.meanCount;
  const relativeCountDiff = countDiff / baseline.meanCount;
  const z = computeZ(baseline.meanDuration, baseline.stdevDuration, current.meanDuration, current.runs);
  const prob = computeProbability(z);
  const isDurationDiffSignificant = prob < PROBABILITY_CONSIDERED_SIGNIFICANT && Math.abs(durationDiff) >= DURATION_DIFF_THRESHOLD_SIGNIFICANT;
  return {
    name,
    baseline,
    current,
    durationDiff,
    relativeDurationDiff,
    isDurationDiffSignificant,
    countDiff,
    relativeCountDiff
  };
}

/**
 * Calculate z-score for given baseline and current performance results.
 *
 * Based on :: https://github.com/v8/v8/blob/master/test/benchmarks/csuite/compare-baseline.py
 */
function computeZ(baselineMean, baselineStdev, currentMean, runs) {
  if (baselineStdev == 0) return 1000;
  return Math.abs((currentMean - baselineMean) / (baselineStdev / Math.sqrt(runs)));
}

/**
 * Compute statisticial hyphotesis probability based on z-score.
 *
 * Based on :: https://github.com/v8/v8/blob/master/test/benchmarks/csuite/compare-baseline.py
 */
function computeProbability(z) {
  // p 0.005: two sided < 0.01
  if (z > 2.575_829) return 0;

  // p 0.010
  if (z > 2.326_348) return 0.01;

  // p 0.015
  if (z > 2.170_091) return 0.02;

  // p 0.020
  if (z > 2.053_749) return 0.03;

  // p 0.025: two sided < 0.05
  if (z > 1.959_964) return 0.04;

  // p 0.030
  if (z > 1.880_793) return 0.05;

  // p 0.035
  if (z > 1.811_91) return 0.06;

  // p 0.040
  if (z > 1.750_686) return 0.07;

  // p 0.045
  if (z > 1.695_397) return 0.08;

  // p 0.050: two sided < 0.10
  if (z > 1.644_853) return 0.09;

  // p 0.100: two sided < 0.20
  if (z > 1.281_551) return 0.1;

  // two sided p >= 0.20
  return 0.2;
}
