// @callstack/reassure-compare/lib/commonjs/utils/format.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.formatChange = formatChange;
exports.formatCount = formatCount;
exports.formatCountChange = formatCountChange;
exports.formatDuration = formatDuration;
exports.formatDurationChange = formatDurationChange;
exports.formatMetadata = formatMetadata;
exports.formatPercent = formatPercent;
exports.formatPercentChange = formatPercentChange;
exports.formatRenderCountChange = formatRenderCountChange;
exports.formatRenderDurationChange = formatRenderDurationChange;
/**
 * Utility functions used for formatting data into strings
 */
function formatPercent(value) {
  const valueAsPercent = value * 100;
  return `${valueAsPercent.toFixed(1)}%`;
}
function formatPercentChange(value) {
  const absValue = Math.abs(value);

  // Round to zero
  if (absValue < 0.005) return `±0.0%`;
  return `${value >= 0 ? '+' : '-'}${formatPercent(absValue)}`;
}
function formatDuration(duration) {
  return `${duration.toFixed(1)} ms`;
}
function formatDurationChange(value) {
  if (value > 0) {
    return `+${formatDuration(value)}`;
  }
  if (value < 0) {
    return `${formatDuration(value)}`;
  }
  return '0 ms';
}
function formatCount(value) {
  return Number.isInteger(value) ? `${value}` : `${value.toFixed(2)}`;
}
function formatCountChange(value) {
  if (value > 0) return `+${value}`;
  if (value < 0) return `${value}`;
  return '±0';
}
function formatChange(value) {
  if (value > 0) return `+${value}`;
  if (value < 0) return `${value}`;
  return '0';
}
function formatRenderDurationChange(entry) {
  const {
    baseline,
    current
  } = entry;
  let output = `${formatDuration(baseline.meanDuration)} → ${formatDuration(current.meanDuration)}`;
  if (baseline.meanDuration != current.meanDuration) {
    output += ` (${formatDurationChange(entry.durationDiff)}, ${formatPercentChange(entry.relativeDurationDiff)})`;
  }
  output += ` ${getRenderDurationSymbols(entry)}`;
  return output;
}
function getRenderDurationSymbols(entry) {
  if (!entry.isDurationDiffSignificant) {
    if (entry.relativeDurationDiff > 0.15) return '🔴';
    if (entry.relativeDurationDiff < -0.15) return '🟢';
    return '';
  }
  if (entry.relativeDurationDiff > 0.33) return '🔴🔴';
  if (entry.relativeDurationDiff > 0.05) return '🔴';
  if (entry.relativeDurationDiff < -0.33) return '🟢🟢';
  if (entry.relativeDurationDiff < -0.05) return ' 🟢';
  return '';
}
function formatRenderCountChange(entry) {
  const {
    baseline,
    current
  } = entry;
  let output = `${formatCount(baseline.meanCount)} → ${formatCount(current.meanCount)}`;
  if (baseline.meanCount != current.meanCount) {
    output += ` (${formatCountChange(entry.countDiff)}, ${formatPercentChange(entry.relativeCountDiff)})`;
  }
  output += ` ${getRenderCountSymbols(entry)}`;
  return output;
}
function getRenderCountSymbols(entry) {
  if (entry.countDiff > 1.5) return '🔴🔴';
  if (entry.countDiff > 0.5) return '🔴';
  if (entry.countDiff < -1.5) return '🟢🟢';
  if (entry.countDiff < -0.5) return '🟢';
  return '';
}
function formatMetadata(metadata) {
  if (metadata !== null && metadata !== void 0 && metadata.branch && metadata !== null && metadata !== void 0 && metadata.commitHash) {
    return `${metadata.branch} (${metadata.commitHash})`;
  }
  return (metadata === null || metadata === void 0 ? void 0 : metadata.branch) || (metadata === null || metadata === void 0 ? void 0 : metadata.commitHash) || '(unknown)';
}
