// @callstack/reassure-compare/lib/commonjs/utils/validate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.parseHeader = parseHeader;
exports.parsePerformanceEntries = parsePerformanceEntries;
exports.performanceMetadataSchema = exports.performanceHeaderSchema = exports.performanceEntrySchema = void 0;
var _zod = require("zod");
var _array = require("./array");
/** Metadata information for performance results. */
const performanceMetadataSchema = _zod.z.object({
  branch: _zod.z.string().optional(),
  commitHash: _zod.z.string().optional()
});

/** Header of performance results file. */
exports.performanceMetadataSchema = performanceMetadataSchema;
const performanceHeaderSchema = _zod.z.object({
  metadata: performanceMetadataSchema
});

/** Entry in the performance results file. */
exports.performanceHeaderSchema = performanceHeaderSchema;
const performanceEntrySchema = _zod.z.object({
  /** Name of the test scenario */
  name: _zod.z.string(),
  /** Number of times the measurment test was run */
  runs: _zod.z.number(),
  /** Arithmetic average of measured render durations for each run */
  meanDuration: _zod.z.number(),
  /** Standard deviation of measured render durations for each run */
  stdevDuration: _zod.z.number(),
  /** Array of measured render durations for each run */
  durations: _zod.z.array(_zod.z.number()),
  /** Arithmetic average of measured render counts for each run */
  meanCount: _zod.z.number(),
  /** Standard deviation of measured render counts for each run */
  stdevCount: _zod.z.number(),
  /** Array of measured render counts for each run */
  counts: _zod.z.array(_zod.z.number())
});
exports.performanceEntrySchema = performanceEntrySchema;
const performanceEntriesSchema = _zod.z.array(performanceEntrySchema).refine(val => !(0, _array.hasDuplicateValues)(val.map(val => val.name)), {
  message: `Your performance result file contains records with duplicated names.
    Please remove any non-unique names from your test suites and try again.`
});
function parseHeader(header) {
  return performanceHeaderSchema.parse(header);
}
function parsePerformanceEntries(entries) {
  return performanceEntriesSchema.parse(entries);
}
